bool appear
int16 cx
int16 cy
float64 degrees
float64 area